src/lib.rs

```diff
@@ -216,6 +293,36 @@ pub unsafe extern "C" fn fw_predict_with_cache(
     predictor.predict_with_cache(str_buffer)
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn fw_predict_with_cache_and_report_stats(
+    ptr: *mut FfiPredictor,
+    input_buffer: *const c_char,
+    stats: *mut f32,
+    stats_size: usize,
+) -> f32 {
+    let str_buffer = c_char_to_str(input_buffer);
+    let predictor: &mut Predictor = from_ptr(ptr);
+    match predictor.predict_with_cache_and_report_stats(str_buffer) {
+        Ok(prediction_stats) => {
+            report_stats(&prediction_stats, stats, stats_size);
+            prediction_stats.mean
+        }
+        Err(e) => e,
+    }
+}
+
+unsafe fn report_stats(prediction_stats: &PredictionStats, stats: *mut f32, stats_size: usize) {
+    if stats_size < 4 {
+        log::error!("Stats buffer is too small, expecting 4, was {}", stats_size);
+        return;
+    }
+    let stats_slice = std::slice::from_raw_parts_mut(stats, stats_size);
+    *stats_slice.get_unchecked_mut(0) = prediction_stats.mean;
+    *stats_slice.get_unchecked_mut(1) = prediction_stats.variance;
```

Comment:
No need to report std and var, var is enough ..

Reply:
We actually need std dev. But I am reporting both just incase we mess up with the calculation.

Reply:
Merely saying that stddev is just sqrt(var), computable more or less everywhere (e.g., during analysis)